tests/unit_tests/test_game.py

```diff
@@ -951,16 +951,76 @@ async def test_game_results(game: Game, players):
     assert len(result_dict["teams"]) == 2
 
     for team in result_dict["teams"]:
-        assert team["outcome"] == (
-            "VICTORY" if team["player_ids"] == [host_id]
-            else "DEFEAT"
-        )
+        outcome, army = ("VICTORY", 0) if team["player_ids"] == [host_id] else ("DEFEAT", 1)
+        assert team["outcome"] == outcome
+        assert team["army_results"] == [
+            ArmyResult(
+                team["player_ids"][0],
+                army,
+                outcome,
+                [],
+            )
+        ]
     assert result_dict["game_id"] == game.id
     assert result_dict["map_id"] == game.map_id
     assert result_dict["featured_mod"] == "faf"
     assert result_dict["sim_mod_ids"] == []
 
 
+async def test_team_game_results(game: Game, game_add_players):
+    game.state = GameState.LOBBY
+    players = (*game_add_players(game, 2, 0), *game_add_players(game, 2, 2))
+
+    await game.launch()
+    for i, player in enumerate(players):
+        for _ in players:
+            if i == 0:  # This player was the last one standing
+                await game.add_result(player.id, i, "victory", 1)
+            else:
+                await game.add_result(player.id, i, "defeat", 1)
+
+    game_results = await game.resolve_game_results()
+    result_dict = game_results.to_dict()
+
+    for i, team in enumerate(result_dict["teams"]):
+        assert team["outcome"] == "VICTORY" if i == 0 else "DEFEAT"
+        assert team["army_results"] == [
+            ArmyResult(
+                player_id,
+                game.get_player_option(player_id, "Army"),
+                "VICTORY" if player_id == 1 else "DEFEAT",
+                [],
+            ) for player_id in team["player_ids"]
+        ]
+
+
+async def test_army_results_present_for_invalid_games(game: Game, game_add_players):
+    game.state = GameState.LOBBY
+    # Team 1 is the FFA_TEAM, will cause the game to have ValidityState.FFA_NOT_RANKED
+    players = (*game_add_players(game, 2, 0), *game_add_players(game, 2, 1))
+
+    await game.launch()
+    for i, player in enumerate(players):
+        if i == 0:  # This player was the last one standing
+            await game.add_result(player.id, i, "victory", 1)
+        else:
+            await game.add_result(player.id, i, "defeat", 1)
+
+    game_results = await game.resolve_game_results()
+    result_dict = game_results.to_dict()
+
+    for i, team in enumerate(result_dict["teams"]):
+        assert team["outcome"] == "UNKNOWN"
```

Comment:
Here's what I was trying to say about gw games. They will most likely be unrated for some reason, but we still want to know what the team outcomes resolve to anyways. Like, even if the game is unrated because it was 3v1, the gw server still needs to know which team won. It would be possible for it to decide based on army results, but then we're just duplicating team outcome resolution across multiple services which isn't great.

Reply:
Should I just remove the `if self.validity is ValidityState.VALID` check entirely then, so the team outcomes are always calculated?

Reply:
I think that would be fine